include/field.hxx

```diff
@@ -121,11 +140,65 @@ class Field {
    */
   virtual int getNz() const;
 
+  friend void swap(Field& first, Field& second) noexcept {
+    using std::swap;
+    swap(first.name, second.name);
+    swap(first.fieldmesh, second.fieldmesh);
+    swap(first.fieldCoordinates, second.fieldCoordinates);
+    swap(first.location, second.location);
+    swap(first.directions, second.directions);
+  }
+
+  friend bool fieldsCompatible(const Field& field1, const Field& field2) {
```

Comment:
Does this need to be a friend? There are getters for the directions, so it would be cleaner to use those